src/components/Compare/CompareMain.tsx

```diff
@@ -103,10 +113,10 @@ const CompareMain = (props: {
     getLocationPageViewMoreCopy(geoScope, countyTypeToView, props.stateName);
   const locationPageLocationsForCompare = getLocationPageLocations();
 
-  const locations = props.isHomepage
+  const locations = !stateId
```

Comment:
I switched to using `!stateId` because to generate the share image I always leave isHomepage set to false since we don't know if compare was shared from the homepage or not.